modules/tinymce/src/core/main/ts/textpatterns/keyboard/KeyHandler.ts

```diff
@@ -9,6 +9,11 @@ import { InlinePatternSet, PatternSet } from '../core/PatternTypes';
 import * as Utils from '../utils/Utils';
 
 const handleEnter = (editor: Editor, patternSet: PatternSet): boolean => {
+  // TINY-8779: mceInsertNewLine calls normalize() which alters the structure of the body content,
+  // in which text nodes of a paragraph are merged into one. This makes the range of any matching patterns become invalid
+  // as the text nodes are no longer existed after normalization. So do a normalize() prior finding any matches
```

Comment:
This description isn't quite correct... `mceInsertNewLine` doesn't call `normalize`, it's the `undoManager.extra()` call that will store the content as a string and then set it back via `setContent()` which results in the fragmented text nodes being merged.